File: HA11_3.cpp
//Accept N numbers from user and accept one another number as NO, return index of last occourence of that NO
//Input : N : 6
//       NO : 66
//        E : 85    66     3    66      93      88
//Output : 3

#include<iostream>
using namespace std;

class Array
{
    public:
        int iSize;
        int iChoice;
        int *Arr;

        Array(int iNo1, int iNo2)
        {
            iSize = iNo1;
            iChoice = iNo2;
            Arr = new int[iSize];
        }

        ~Array()
        {
            delete []Arr;
        }

        void Accept()
        {
            int iCnt = 0;

            cout<<"Enter Elements : "<<"\n";

            for(iCnt =0 ; iCnt < iSize; iCnt++)
            {
                cin>>Arr[iCnt];
            }
        }

        void Display()
        {
            int iCnt = 0;

            cout<<"Elements are : "<<"\n";

            for(iCnt = 0; iCnt < iSize; iCnt++)
            {
                cout<<Arr[iCnt]<<"\n";
            }
        }

        int FirstOccourance()
        {
            int iCnt = 0;
            int index = -1;
            int iCount = 0;

            for(iCnt =0; iCnt < iSize; iCnt++)
            {
                iCount ++;

                if(Arr[iCnt] == iChoice)
                {
                    index = iCnt;
                    continue;
                }
            }
            return index;
        }
};

int main()
{
    int iNo1 = 0;
    int iNo2 = 0;
    int iRet = 0;

    cout<<"Enter Number of Elements : "<<"\n";
    cin>>iNo1;

    cout<<"Enter Number you want to check : "<<"\n";
    cin>>iNo2;

    Array aobj(iNo1, iNo2);
    aobj.Accept();
    aobj.Display();
    iRet = aobj.FirstOccourance();

    cout<<"Index is : "<<iRet<<"\n";

    return 0;
}

File: HA11_3.java
import java.util.Scanner;

class ArrayX
{
    public int Arr[];
    public int No2;

    ArrayX(int iSize, int iChoice)
    {
        Arr = new int [iSize];
        this.No2 = iChoice;
    }

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        int iCnt = 0;

        System.out.println("Enter the Elements : ");

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            Arr[iCnt] = sobj.nextInt();
        }
    }

    public void Display()
    {
        Scanner sobj = new Scanner(System.in);
        int iCnt = 0;

        System.out.println("Enter the Elements : ");

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            System.out.println(Arr[iCnt]);
        }
    }

    public int ChkIndex()
    {
        Scanner sobj = new Scanner(System.in);
        int iCnt = 0;
        int index = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if(Arr[iCnt] == No2)
            {
                index = iCnt;
                continue;
            }
        }
        return index;
    }
}

class HA11_4
{
    public static void main(String[] args) 
    {
        int iRet = 0;
        
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the Number of Elements : ");
        int iSize = sobj.nextInt();

        System.out.println("Enter the Number you want to check : ");
        int iNo2 = sobj.nextInt();

        ArrayX aobj = new ArrayX(iSize, iNo2);
        aobj.Accept();
        aobj.Display();
        iRet = aobj.ChkIndex();

        System.out.println("Index is : "+ iRet);

        sobj.close();
    }
}
